### DISCUSSION TITLE
What is the purpose of the one byte packet?

### DISCUSSION THREAD

**Discussion by ghost** [esnet/iperf#1307] [Q&A]
I'm running iPerf 3.1.3 and I noticed that before sending the settings json string (in TCP), iPerf sends a single byte packet with the value of `9`.

What is that exactly?

**Comment by bmah888** [comment]
You're seeing a control protocol for the client and server to exchange state, start/stop the test, get results, etc.

`9` is the code for exchanging test parameters and it's followed by the JSON string. The various protocol states are defined in `iperf_api.h` (look for `PARAM_EXCHANGE`). This binary protocol isn't very well-documented, unfortunately.

**Reply by ghost** [reply]
Why would the server respond with this control message anyway if it's always the next step? (Exchanging params)
